### ISSUE TITLE
Feture request: Task management with notes

### ISSUE THREAD

**Issue by sharpevo** [pbek/QOwnNotes#550]
Thanks for the great app and the hard efforts behind it, so that I can get the updates everyday ;-)

Here's a feature request about task management along with notes. Most of time, I need to get the note list which contains tasks, in order to follow the notes that to be continued more elagently. The dialog look like:

![note](https://cloud.githubusercontent.com/assets/7692030/23977091/ee22cbfa-0a26-11e7-9566-33acf9c896ef.png)

The features are including:
- Get the status(active, completed, etc) from "Tasks" app of nextcloud
- The task entry is clickable, and open the cooresponding note.

I was trying to get it done by scripts, but there's only `fetchNoteByFileName`, rather than functions like `fetchNoteByText`, so that I can search strings like `(task://...)`.

Thank you!

**Comment by pbek** [comment]
Thank you for your kind words and your suggestion.
I think I'm not sure what you want. You write about a note list but your screenshots shows a task list. What are you trying to achieve?

**Comment by sharpevo** [comment]
Actually it's a list of notes that holds tasks, for example, the notes and tasks are:

- note 1, contains task 1 and task 3.
- note 2, contains task2 only.

So I think the following list is more convenient to manage tasks:

- task 1 [note 1]
- task 2 [note 2]
- task 3 [note 1]

As the implementation, I think the following two steps are necessary:

1. select notes where "(task://" is in the note text.
2. build the task-oriented data structure.

Thank you!

**Comment by pbek** [comment]
thank you for the explanation

**Comment by QuAzI** [comment]
GitHub and TortoiseHG can use `#number` like issue links. As example #550 is link to this issue. Can it be used also in QOwnNotes? But issue tracker must be configurable. 

Mercurial (Tortoise HG) settings example
![2017-07-14_23-27-58_vl28ej0l](https://user-images.githubusercontent.com/1897605/28229645-3978103c-68ec-11e7-9881-044106e6c714.png)

**Comment by pbek** [comment]
Nah, I don't think that will be possible. Working with non-conform urls a is VERY hard thing to deal with...
